pygmt/src/rose.py

```diff
@@ -0,0 +1,245 @@
+         all four labels. Note that the :gmt-term:`GMT_LANGUAGE` setting
+         will affect the words used.
+
+    no_scale : bool
+         Do NOT draw the scale length bar (``no_scale = True``).
+         Default plots scale in lower right corner provided **frame**
+         is used.
+
+    shift : bool
+         Shift sectors so that they are centered on the bin interval
+         (e.g., first sector is centered on 0 degrees).
+
+    vectors : str
+          ``vectors = 'mode_file'``. Plot vectors showing the
+          principal directions given in the *mode_file* file.
+          Alternatively, specify **vectors** to compute and plot
+          mean direction. See **vector_params** to control the vector
+          attributes. Finally, to instead save the computed mean
+          direction and other statistics, use
+          ``vectors = '+wmode_file'``. The eight items saved to
+          a single record are: *mean_az*, *mean_r*, *mean_resultant*,
+          *max_r*, *scaled_mean_r*, *length_sum*, *n*, *sign@alpha*,
+          where the last term is 0 or 1 depending on whether the mean
+          resultant is significant at the level of confidence set via
+          **conf_level**.
+
+    vector_params : str
+        Used with **vectors** to modify vector parameters. For
+        vector heads, append vector head size [Default is 0, i.e., a
+        line]. See :gmt-docs:`rose.html#vector-attributes` for
+        specifying additional attributes. If **vectors** is not
+        given and the current plot mode is to draw a windrose diagram
+        then using **vector_params** will add vector heads to all
+        individual directions using the supplied attributes.
+
+    alpha : float or str
+        Sets the confidence level used to determine if the mean
+        resultant is significant (i.e., Lord Rayleigh test for
+        uniformity) [``alpha = 0.05``]. Note: The critical
+        values are approximated [Berens, 2009] and requires at
+        least 10 points; the critical resultants are accurate to at
+        least 3 significant digits. For smaller data sets you
+        should consult exact statistical tables.
+
+        Berens, P., 2009, CircStat: A MATLAB Toolbox for Circular
+        Statistics, *J. Stat. Software*, 31(10), 1-21,
+        https://doi.org/10.18637/jss.v031.i10.
+
+    inquire : bool
+        Inquire. Computes statistics needed to specify a useful
+        **frame**. No plot is generated. The following statistics
+        are written to stdout: *n*, *mean az*, *mean r*, *mean
+        resultant length*, *max bin sum*, *scaled mean*, and
+        *linear length sum*.
+
+    {U}
+    {V}
+    {XY}
+    {p}
+    {t}
+    """
+
+    kwargs = self._preprocess(**kwargs)
+
+    kind = data_kind(data, length, azimuth)
+
+    extra_arrays = []
+    if "t" in kwargs and is_nonstr_iter(kwargs["t"]):
+        extra_arrays.append(kwargs["t"])
+        kwargs["t"] = ""
```

Comment:
I don't test it, but I think varying transparency is not available for `rose`.

Reply:
@seisman you're right, I removed that passage from the code.